datalad/support/tests/test_fileinfo.py

```diff
@@ -280,3 +280,35 @@ def test_get_content_info_dotgit(path):
     # Files in .git/ won't be reported, though this takes a kludge on our side
     # before Git 2.25.
     assert_false(ds.repo.get_content_info(paths=[op.join(".git", "config")]))
+
+
+@with_tempfile
+def test_get_content_info_paths_empty_list(path):
+    ds = Dataset(path).create()
+
+    # Unlike None, passing any empty list as paths to get_content_info() does
+    # not report on all content.
+    assert_false(ds.repo.get_content_info(paths=[]))
+    assert_false(ds.repo.get_content_info(paths=[], ref="HEAD"))
+
+    # Add annex content to make sure its not reported.
+    (ds.pathobj / "foo").write_text("foo")
+    ds.save()
+
+    # Same for get_content_annexinfo()...
+    assert_false(ds.repo.get_content_annexinfo(paths=[]))
+    assert_false(ds.repo.get_content_annexinfo(paths=[], init=None))
+    assert_false(ds.repo.get_content_annexinfo(paths=[], ref="HEAD"))
+    assert_false(
+        ds.repo.get_content_annexinfo(paths=[], ref="HEAD", init=None))
+    # ... where whatever was passed for init will be returned as is.
+    assert_equal(
+        ds.repo.get_content_annexinfo(
+            paths=[], ref="HEAD", init={"random": {"entry": "a"}}),
+        {"random": {"entry": "a"}})
+
+
+@with_tempfile
+def test_status_paths_empty_list(path):
+    ds = Dataset(path).create()
+    assert_false(ds.repo.status(paths=[]))
```

Comment:
```suggestion
    assert_equal(ds.repo.status(paths=[]), [])
```
to make explicit and not wonder if could be `None` (or is it `None`?)?

Reply:
It's an empty `OrderedDict`.  I've updated the test.